posthog/warehouse/sync_resource.py

```diff
@@ -50,3 +54,39 @@ def _sync_resource(resource_id):
             resource.are_tables_created = True
             resource.status = job["status"]
             resource.save()
+
+
+def get_rows_synced_by_team(begin: datetime, end: datetime, team_id):
+    resources = ExternalDataSource.objects.filter(team_id=team_id, are_tables_created=True)
+    return sum([get_rows_synced_by_resource_id(begin, end, resource.pk) for resource in resources])
+
+
+def get_rows_synced_by_resource_id(begin: datetime, end: datetime, resource_id, offset=0):
+
+    resource = ExternalDataSource.objects.get(pk=resource_id)
+    params = {
+        "connectionId": resource.connection_id,
+        "limit": 100,
+        "offset": offset,
+        "status": "succeeded",
+        "updatedAtStart": begin.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "updatedAtEnd": end.strftime("%Y-%m-%dT%H:%M:%SZ"),
+    }
+
+    return _accumulate_jobs_field(AIRBYTE_JOBS_URL + "?" + urlencode(params), "rowsSynced")
+
+
+def _accumulate_jobs_field(url, field, acc=0):
```

Comment:
oooh something worth thinking about:

The usage report can be fragile, and this introduces another external dependency, which can bork things for us.

Also not sure about the pagination limits here, and how long these requests take, do they make a scan query in airbyte over time? Or is it a stored value that can be quickly returned? Will we need to back-off retries if too many teams made too many syncs? Is there a way to get this for multiple `connectionId`s in one request?

(I have zero context here, so take it with a pinch of salt): Depending on fragility, it's worth considering that:

1. We can switch this off on demand so default usage report isn't affected
2. We can move this into it's own accumulation job, that either updates & sends events to team 2 when row syncs happen (`decide` does this too for requests)
3. It's reliable & fast enough to continue as part of usage report

Reply:
Good points thanks for the review. Was thinking of moving it to something like 2